# Introdução

# Modelo

##  analise dos dados

In [2]:
import pandas as pd

In [8]:
df = pd.read_csv("../datasets/credit_01/test")
df.head()

REF_DATE  TARGET VAR2   IDADE VAR4 VAR5       VAR6  \
0  2017-01-04 00:00:00+00:00       0    M  61.532  NaN   BA -12.600989   
1  2017-08-22 00:00:00+00:00       0    F  28.586  NaN   PE  -7.854283   
2  2017-01-04 00:00:00+00:00       1  NaN     NaN  NaN   MG -17.217517   
3  2017-01-09 00:00:00+00:00       1  NaN     NaN  NaN   PA  -1.884456   
4  2017-05-11 00:00:00+00:00       0    M  24.288  NaN   SP -23.612733   

        VAR7 VAR8 VAR9  ...       VAR141  VAR142  VAR143  VAR144  VAR145  \
0 -38.965939  NaN    E  ...  2380.533735       D     NaN     NaN     NaN   
1 -35.763047    E    E  ...  1428.485398       E     NaN     NaN     NaN   
2 -46.872278  NaN    E  ...  1428.485398       E     NaN     NaN     NaN   
3 -48.765531  NaN    E  ...  1428.485398       E     NaN     NaN     NaN   
4 -46.762968    E    E  ...  1593.518909       E     NaN     NaN     NaN   

   VAR146  VAR147                                             VAR148  VAR149  \
0     NaN     102                                  EMAIL INEXISTENTE   2.6.1   
1     NaN     102                                  EMAIL INEXISTENTE   2.6.1   
2     NaN     102               NOME INEXISTENTE#@#EMAIL INEXISTENTE   2.6.1   
3     NaN     102  EMAIL INEXISTENTE#@#NOME INEXISTENTE#@#CEP INE...   2.6.1   
4     NaN     102                                  EMAIL INEXISTENTE   2.6.1   

       ID  
0    8396  
1   74636  
2   95855  
3  120857  
4  126978  

[5 rows x 151 columns]

Os dados contem dados bastante heterogeneos. Observamos strings, inteiros, caracteres e etc... sem serem propriamente indentificados. Primeiramente separemos os dados compostos por numeros e outros tipos de dados.

In [24]:
df_num = pd.DataFrame()
df_obj = pd.DataFrame()
for colum in df.columns:
    if df[colum].dtype!="object":
        df_num[colum]= df[colum]
    else:
        df_obj[colum]= df[colum]
for colum in df_num.columns:
    print(df[colum].dtype)

print("******************************************************************************")
for colum in df_obj.columns:
    print(df[colum].dtype)



int64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
float64
int64
int64
******************************************************************************
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
object
o

/tmp/ipykernel_15820/3235514565.py:7: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df_obj[colum]= df[colum]
/tmp/ipykernel_15820/3235514565.py:7: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df_obj[colum]= df[colum]
/tmp/ipykernel_15820/3235514565.py:7: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
 

In [25]:
print("num of number "+ str(df_num.columns.size))
print("num of objects "+ str(df_obj.columns.size))

num of number 37
num of objects 114


vemos que o numero de dados composto por numeros é 37, e o de objetos são 114

In [26]:
df_num.to_csv("nums_param.csv")
df_obj.to_csv("obj_param.csv")

# ler texto do capitulo 3 e 5 do brownlee , "statistical imputation" e "how to encode categorical data"